hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/precompile/impl/TokenCreatePrecompile.java

```diff
@@ -605,6 +672,38 @@ public static TokenCreateWrapper decodeTokenCreateWithoutFees(
                 tokenExpiry);
     }
 
+    public static TokenCreateWrapper decodeTokenCreateWithoutFeesV2(
+            @NotNull final Tuple tokenCreateStruct,
+            final boolean isFungible,
+            final BigInteger initSupply,
+            final BigInteger decimals,
+            final UnaryOperator<byte[]> aliasResolver) {
+        final var tokenName = (String) tokenCreateStruct.get(0);
+        final var tokenSymbol = (String) tokenCreateStruct.get(1);
+        final var tokenTreasury =
+                convertLeftPaddedAddressToAccountId(tokenCreateStruct.get(2), aliasResolver);
+        final var memo = (String) tokenCreateStruct.get(3);
+        final var isSupplyTypeFinite = (Boolean) tokenCreateStruct.get(4);
+        final var maxSupply = (long) tokenCreateStruct.get(5);
+        final var isFreezeDefault = (Boolean) tokenCreateStruct.get(6);
+        final var tokenKeys = decodeTokenKeys(tokenCreateStruct.get(7), aliasResolver);
+        final var tokenExpiry = decodeTokenExpiryV2(tokenCreateStruct.get(8), aliasResolver);
```

Comment:
You can try passing `decodeTokenExpiryV2` as a lambda or think of another approach to keep a single` `decodeTokenCreateWithoutFees()` method.